lib/api/backbeat/listLifecycleCurrents.js

```diff
@@ -37,11 +38,20 @@ function listLifecycleCurrents(authInfo, request, log, callback) {
         Number.parseInt(params['max-keys'], 10) : 1000;
     if (Number.isNaN(requestMaxKeys) || requestMaxKeys < 0) {
         monitoring.promMetrics(
-            'GET', bucketName, 400, 'listBucket');
+            'GET', bucketName, 400, 'listLifecycleCurrents');
         return callback(errors.InvalidArgument);
     }
     const actualMaxKeys = Math.min(constants.listingHardLimit, requestMaxKeys);
 
+    const excludedDataStoreName = params['excluded-data-store-name'];
+    if (excludedDataStoreName && !locationConstraints[excludedDataStoreName]) {
+        const errMsg = getLocationConstraintErrorMessage(excludedDataStoreName);
+        log.error(`locationConstraint is invalid - ${errMsg}`, { locationConstraint: excludedDataStoreName });
+        monitoring.promMetrics('GET', bucketName, 400, 'listLifecycleCurrents');
+
+        return callback(errors.InvalidLocationConstraint.customizeDescription(errMsg));
+    }
```

Comment:
This check seems to be duplicated in each listing backend that uses it, why not instead validating in `listLifecycle`, before calling the backend function? This would also avoid having to pass the `locationConstraints` object around.

Reply:
List orphans delete markers don't use it.

Reply:
The check is utilized by both the 'listLifecycleCurrents' and 'listLifecycleNonCurrents' API calls, but not by the 'listLifecycleDeleteMarkers' API call. To clarify that the check applies only to the former two calls, I moved it to the 'API level'.

Reply:
Well, still not very clear to me ;)

I think for separation of concerns, it's better if the check on the locations happens from the API caller. It's ok if it is also checked for the "listLifecycleDeleteMarkers" case. Then, we pass the excluded (and validated) location to the backend listing, and it's up to the listing to decide whether to implement this exclusion mechanism or not (and document appropriately in JSDoc).

I believe it works as is but I find it a bit confusing and less easily maintainable.

Reply:
For "listLifecycleDeleteMarkers" case, I think we should check that `params['excluded-data-store-name']` is `undefined` instead and we should return a different error than "getLocationConstraintErrorMessage" if `params['excluded-data-store-name']` is defined.

I have not implemented this check today since it is an internal API but I will create a ticket for it.